eng/cmake/global_compile_options.txt

```diff
@@ -15,5 +15,5 @@ else()
     set(WARNINGS_AS_ERRORS_FLAG "-Werror")
   endif()
 
-  add_compile_options(-Wall -Wextra -pedantic  ${WARNINGS_AS_ERRORS_FLAG})
+  add_compile_options(-Wall -Wextra -pedantic  ${WARNINGS_AS_ERRORS_FLAG} -Wunused -Wuninitialized -Wmissing-declarations -Wconversion -Wpointer-arith -Wshadow -Wlogical-op -Waggregate-return -Wfloat-equal)
```

Comment:
We should scrub all of the warnings in this file and use global_compile_options.txt instead of duplicating logic/flags down through all of the CMakeLists.txt files. It'll keep things better synchronized when we add/remove flags.

Reply:
Thank you Daniel, I was going to ask you, I will remove the duplication now!